### PR TITLE
More improvements for the crate publishing pipeline

### PR DESCRIPTION
**Pull request by joao-paulo-parity**

### CODE REVIEW

File: scripts/ci/gitlab/pipeline/publish.yml [hunk 1]
```diff
@@ -189,18 +189,18 @@ publish-draft-release:
   # the releases of one another. Use resource_group to ensure that at most one instance of this job
   # is running at any given time.
   resource_group:                  crates-publishing
-  variables:
-    # crates.io rate limits crates publishing by 1 per minute, so a delay needs to be inserted
-    # slightly higher than that after publishing each crate. The value is specified in seconds.
-    SPUB_AFTER_PUBLISH_DELAY: 64
```
Comment (joao-paulo-parity) [inline scripts/ci/gitlab/pipeline/publish.yml:195]: This delay is part of an implementation detail and thus it has been moved to the `publish-crates` script itself.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -251,9 +241,7 @@ variables:
     paths:
       - artifacts/
   variables:
-    CRATESIO_API: https://crates.io/api
     CRATESIO_CRATES_OWNER: parity-crate-owner
-    GH_API: https://api.github.com
```
Comment (joao-paulo-parity) [inline .gitlab-ci.yml:256]: Those defaults have been moved to the publish-crates script

File: scripts/ci/gitlab/pipeline/publish.yml [hunk 3]
```diff
@@ -189,18 +189,18 @@ publish-draft-release:
   # the releases of one another. Use resource_group to ensure that at most one instance of this job
   # is running at any given time.
   resource_group:                  crates-publishing
-  variables:
```
Comment (joao-paulo-parity) [inline scripts/ci/gitlab/pipeline/publish.yml:192]: Placeholder comment to trigger GitHub Action

**Review by alvicsam — APPROVED** [review]

**Review by sergejparity — APPROVED** [review]

**Comment by joao-paulo-parity** [comment]
bot merge

**Pull request merged**